mhctools/cli/script.py

```diff
@@ -53,37 +61,51 @@ def add_output_args(parser):
 add_input_args(arg_parser)
 add_output_args(arg_parser)
 
-def main(args_list=None):
-    """
-    Script to make pMHC binding predictions from amino acid sequences.
-
-    Usage example:
-        mhctools
-            --sequence SFFPIQQQQQAAALLLI \
-            --sequence SILQQQAQAQQAQAASSSC \
-            --mhc-predictor netmhc \
-            --mhc-alleles HLA-A0201 H2-Db \
-            --output-csv epitope.csv
-    """
+def parse_args(args_list=None):
     if args_list is None:
         args_list = sys.argv[1:]
-    args = arg_parser.parse_args(args_list)
+    return arg_parser.parse_args(args_list)
+
+def run_predictor(args):
     predictor = mhc_binding_predictor_from_args(args)
 
     if args.input_fasta_file:
         input_dictionary = parse_fasta_dictionary(args.input_fasta_file)
         binding_predictions = predictor.predict_subsequences(input_dictionary)
     elif args.sequence:
-        binding_predictions = predictor.predict_peptides(args.sequence)
+        if args.extract_subsequences:
+            binding_predictions = predictor.predict_subsequences(args.sequence)
+        else:
+            binding_predictions = predictor.predict_peptides(args.sequence)
     elif args.input_peptides_file:
         with open(args.input_peptides_file) as f:
-            peptides = [line.strip() for line in f]
+            peptides = [line.strip() for line in f if line]
+        if args.extract_subsequences:
```

Comment:
also nit: indent these once more so that the use of `peptides` is in the scope of the context manager?